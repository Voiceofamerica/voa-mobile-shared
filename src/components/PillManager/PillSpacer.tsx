
import * as React from 'react'

import {
  divider,
} from './PillManager.scss'

export interface Props {
  className?: string
  style?: React.CSSProperties
}

class PillSpacer extends React.Component<Props> {
  render () {
    const { className = '', style = {}, children } = this.props

    return (
      <div className={`${divider} ${className}`} style={style}>
        {children}
      </div>
    )
  }
}

export default PillSpacer
