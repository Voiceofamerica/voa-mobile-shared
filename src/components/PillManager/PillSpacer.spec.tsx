
import * as React from 'react'
import { create } from 'react-test-renderer'

import PillSpacer from './PillSpacer'

describe(`<${PillSpacer.name} />`, () => {
  describe('shapshots', () => {
    it('should render without children', () => {
      const element = create((
        <PillSpacer />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with children', () => {
      const element = create((
        <PillSpacer>
          This should show up
        </PillSpacer>
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with className', () => {
      const element = create((
        <PillSpacer className='myClass' />
      ))

      expect(element).toMatchSnapshot()
    })
    it('should render with style', () => {
      const element = create((
        <PillSpacer style={{ background: 'myColor' }} />
      ))

      expect(element).toMatchSnapshot()
    })
  })
})
