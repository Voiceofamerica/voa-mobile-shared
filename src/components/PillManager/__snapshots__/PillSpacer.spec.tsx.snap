// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PillSpacer /> shapshots should render with children 1`] = `
<div
  className="divider "
  style={Object {}}
>
  This should show up
</div>
`;

exports[`<PillSpacer /> shapshots should render with className 1`] = `
<div
  className="divider myClass"
  style={Object {}}
/>
`;

exports[`<PillSpacer /> shapshots should render with style 1`] = `
<div
  className="divider "
  style={
    Object {
      "background": "myColor",
    }
  }
/>
`;

exports[`<PillSpacer /> shapshots should render without children 1`] = `
<div
  className="divider "
  style={Object {}}
/>
`;
